geoportailv3/static/js/shorturlservice.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * @fileoverview This file provides an Angular service for interacting
+ * with the "shorturl" web service.
+ */
+goog.provide('app.GetShorturl');
+
+goog.require('app');
+
+
+/**
+ * @typedef {function(ol.Coordinate=):!angular.$q.Promise}
+ */
+app.GetShorturl;
+
+
+/**
+ * @param {angular.$http} $http The Angular $http service.
+ * @param {ngeo.Location} ngeoLocation The ngeo Location service.
+ * @param {string} shorturlServiceUrl The URL to the "shorturl" service.
+ * @return {app.GetShorturl} The getShorturl function.
+ * @private
+ * @ngInject
+ */
+app.getShorturl_ = function($http, ngeoLocation, shorturlServiceUrl) {
+  return (
+      /**
+     * @param {ol.Coordinate=} opt_coordinate
+     * @return {!angular.$q.Promise} Promise providing the short URL.
+     */
+      function getShorturl(opt_coordinate) {
+        var location = ngeoLocation;
+        if (opt_coordinate) {
+          location.updateParams({
+            'X': Math.round(opt_coordinate[0]),
+            'Y': Math.round(opt_coordinate[1])
+          });
+        }
+        var url = location.getUriString();
+        return $http.get(shorturlServiceUrl, {
+          params: {
+            'url': url
+          }
+        }).then(
+            /**
```

Comment:
I've had bad experience with fixjsstyle and I never use it.

On Tue, Jun 2, 2015 at 4:20 PM, Patrick Weber <notifications@github.com>
wrote:

> In geoportailv3/static/js/shorturlservice.js
> <https://github.com/Geoportail-Luxembourg/geoportailv3/pull/507#discussion_r31527097>
> :
>
> > +     */
> > +      function getShorturl(opt_coordinate) {
> > +        var location = ngeoLocation;
> > +        if (opt_coordinate) {
> > +          location.updateParams({
> > +            'X': Math.round(opt_coordinate[0]),
> > +            'Y': Math.round(opt_coordinate[1])
> > +          });
> > +        }
> > +        var url = location.getUriString();
> > +        return $http.get(shorturlServiceUrl, {
> > +          params: {
> > +            'url': url
> > +          }
> > +        }).then(
> > +            /**
>
> fixjsstyle creates this indent, seems like a bug
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Geoportail-Luxembourg/geoportailv3/pull/507/files#r31527097>
> .
>



-- 
Eric Lemoine

Camptocamp France SAS
Savoie Technolac, BP 352
73377 Le Bourget du Lac, Cedex

Tel : +33 (0)4 58 48 20 34
Mail : eric.lemoine@camptocamp.com
http://www.camptocamp.com